src/App.js

```diff
@@ -5,7 +5,37 @@ import Slider from 'rc-slider';
 import './App.css';
 import 'rc-slider/assets/index.css';
 
+function generateInitialGridState({ width = 1, height = 1}) {
+  let id = 0;
+  const outerMatrix = [];
+
+  for (let i = 0; i < width; i++) {
+    const innerMatrix = [];
+    for (let j = 0; j < height; j++) {
+      innerMatrix.push({ id, state: 'active' });
+      id++;
+    }
+    outerMatrix.push(innerMatrix);
+  }
+
+  return outerMatrix;
+}
+
 class App extends Component {
+  state = {
+    gridLayout: generateInitialGridState({ width: 8, height: 3 })
+  }
+
+  toggleButton = buttonNumber => this.setState(previous => {
```

Comment:
This is a bit of a fiddly way to do it but all this does is update the corresponding spot in the model to the button being pressed.